### ISSUE TITLE
Make analyzers for missing flags & types default disabled?

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/ReflectionAnalyzers#132]
Flags & types is noisy in the code and we have analyzers checking for ambiguous match. Default disabled means they would be opt in.

We could even write analyzers checking if flags and types are specified when not needed and fixes to remove.

**Comment by jnm2** [comment]
I really think it's a code quality issue to leave them out. It's (historically very) risky, and the tradeoff is making reflection look a bit uglier than it already does.

What about an analyzer that offers creation of a `const` to hold the particular combination of flags?

**Comment by JohanLarsson** [comment]
I'm not a fan of the `const`, especially if it is a field, I value code locality highly, to be able to see what is going on.
We can still add it of course but that is a separate issue.

Note that we will not remove the analyzers we have written for flags & types.

> It's (historically very) risky
Those were the old days, now we have analyzers checking things so the risk should go down significantly.

**Comment by jnm2** [comment]
@JohanLarsson 

> Those were the old days, now we have analyzers checking things so the risk should go down significantly.

Only as long as the code doesn't move to a project that doesn't have your analyzer, ad only as long as it runs against the same assemblies at runtime that it is compiled against (which in the case of the BCL it never does).

**Comment by JohanLarsson** [comment]
So maybe best to keep things as is and make light mode opt-in if we bother to write it.

**Comment by jnm2** [comment]
Yeah; I'm interested to hear other opinions, but my current way of thinking is that use of those other overloads is a correctness problem.
